evaluation/tasks/auto_task.py

```diff
@@ -1,25 +1,68 @@
 from abc import ABC, abstractmethod
+from typing import Dict
 import os
 
-from evaluation.utils.io import save_json
+from transformers import AutoTokenizer
+
+from evaluation.utils.io import save_json, load_json
+from evaluation.models import load_model
 
 
 class AutoTask(ABC):
-    def __init__(self, tokenizer, model, device):
-        self.tokenizer = tokenizer
+    def __init__(
+        self, 
+        device, 
+        is_english_only: bool, 
+        model=None, 
+        tokenizer=None,
+        model_name_or_path="",  
+        tokenizer_name="",
+    ):
+        assert model or model_name_or_path, "Expected either `model` or `model_name_or_path`"
+        assert (
+            tokenizer or tokenizer_name or model_name_or_path
+        ), "Expected either `tokenizer` or `model_name_or_path` or `tokenizer_name`"
+        if model is None:
+            model = load_model(model_name_or_path).to(device)
         self.model = model
+        if tokenizer is None:
+            tokenizer = AutoTokenizer.from_pretrained(tokenizer_name or model_name_or_path)
+        self.tokenizer = tokenizer
         self.device = device
         self.metrics = {}
+        self.task_config = self.load_task_args(is_english_only)
 
     @classmethod
-    def from_task_name(cls, task_name: str, tokenizer, model, device):
+    def from_task_name(
+        cls, 
+        task_name: str, 
+        device, 
+        is_english_only: bool, 
+        model=None, 
+        tokenizer=None,
+        model_name_or_path="",  
+        tokenizer_name="",
+    ):
         all_tasks = cls.__subclasses__()
         for task in all_tasks:
             if task.get_display_name() == task_name:
-                return task(tokenizer=tokenizer, model=model, device=device)
+                return task(
+                    device=device, 
+                    is_english_only=is_english_only,
+                    model=model, 
+                    tokenizer=tokenizer,
+                    model_name_or_path=model_name_or_path,  
+                    tokenizer_name=tokenizer_name,
+                )
         
         raise ValueError(f'Invalid task: {task_name}')
 
+    def load_task_args(self, is_english_only) -> Dict:
+        task_root = os.path.join("evaluation", "tasks", self.get_display_name())        
+        if is_english_only:
+            return load_json(os.path.join(task_root, "english.json"))
+        return load_json(os.path.join(task_root, "multiligual.json"))
```

Comment:
nit: simplify logic a bit
```suggestion
        config_filename =  "english.json" if is_english_only else "multiligual.json"
        return load_json(os.path.join(task_root, config_filename))
```